airflow/providers/google/cloud/hooks/compute_ssh.py

```diff
@@ -191,6 +197,65 @@ def intify(key, value, default):
             if self.cmd_timeout is NOTSET:
                 self.cmd_timeout = CMD_TIMEOUT
 
+    def _check_running_in_parallel(self) -> bool:
+        """
+        Helper function to determine if there are several tasks that are running in parallel.
+
+        :return: boolean, True if there are more than 1 task running at the same time
+        """
+        dag_id = self.context["dag"].dag_id  # type: ignore
+        running_tasks = []
+        with create_session() as session:
+            for task_instance in (
+                session.query(TaskInstance)
+                .filter(TaskInstance.state == State.RUNNING, TaskInstance.dag_id == dag_id)
+                .all()
+            ):
+                running_tasks.append(task_instance)
```

Comment:
It's just a matter of style if the number of tasks isn't huge, but do you think it would be a more pythonic way if we replace the loop by a set or list comprehension?

```python
with create_session() as session:
    query = session.query(TaskInstance)
        .filter(TaskInstance.state == State.RUNNING, TaskInstance.dag_id == dag_id)
        .all()
    running_tasks = set(query)
```

Reply:
Good idea, thank you :)